tests/test_parallel_embedding_composite.py

```diff
@@ -0,0 +1,568 @@
+        for embeddings in [embeddings0, None]:  # Provided or searched
+            sampler = ParallelEmbeddingComposite(
+                mock_sampler, embeddings=embeddings, source=source
+            )
+            num_reads = 2
+            ss = sampler.sample_ising(h, J, num_reads=num_reads)
+            if embeddings is None:
+                self.assertEqual(
+                    num_reads * len(sampler.embeddings), sum(ss.record.num_occurrences)
+                )
+            else:
+                self.assertEqual(
+                    num_reads * len(embeddings), sum(ss.record.num_occurrences)
+                )
+            self.assertTrue(np.all(ss.record.energy == -1.75))
+            self.assertTrue(np.all(ss.record.sample == -1))
+
+    def test_composite_propagation(self):
+        # Propagation fails for TilingComposite but succeeds here.
+        # When using find_sublattice_embedding it is necessayr to specify
+        # the family and shape of the QPU as part of embedder_kwargs
+        mock_sampler0 = MockDWaveSampler()
+        mock_sampler = SpinReversalTransformComposite(mock_sampler0)
+        embeddings = []
+        used_nodes = set()
+        for e in mock_sampler.child.edgelist:
+            if e[0] not in used_nodes and e[1] not in used_nodes:
+                used_nodes.add(e[0])
+                used_nodes.add(e[1])
+                embeddings.append({idx: (n,) for idx, n in enumerate(e)})
+        sampler = ParallelEmbeddingComposite(mock_sampler, embeddings=embeddings)
+        source = tile = dnx.chimera_graph(1, 1, 4)  # A 1:1 mapping assumed
+        J = {e: -1 for e in tile.edges}  # A ferromagnet on the Chimera tile.
+        embedder_kwargs = {"max_num_emb": None}
+        sampler = ParallelEmbeddingComposite(
+            mock_sampler, source=source, embedder_kwargs=embedder_kwargs
+        )
+        sampleset = sampler.sample_ising({}, J, num_reads=1)
+        self.assertGreater(
+            len(sampleset), 1
+        )  # Equal to the number of parallel embeddings
+        
+        embedder = find_sublattice_embeddings
+        embedder_kwargs = {
+            "max_num_emb": None,
+            "tile": tile,
+            "T_family": mock_sampler0.properties["topology"]["type"],
+            "T_kwargs": {"m": mock_sampler0.properties["topology"]["shape"][0]},
+        }
+        try:
+            sampler = ParallelEmbeddingComposite(
+                mock_sampler,
+                source=source,
+                embedder=embedder,
+                embedder_kwargs=embedder_kwargs,
+            )
+            sampleset = sampler.sample_ising({}, J, num_reads=1)
+            self.assertGreater(
+                len(sampleset), 1
+            )  # Equal to the number of parallel embeddings
+        except:
+            print('Test is skipped for now, relies on incomplete pull request:'
+                  'https://github.com/jackraymond/minorminer/tree/parallel_embeddings')
```

Comment:
Is that PR ready by any chance?

Reply:
https://github.com/dwavesystems/minorminer/pull/266